rust/src/composepost.rs

```diff
@@ -344,13 +344,18 @@ fn compose_postprocess_units(rootfs_dfd: &Dir, treefile: &mut Treefile) -> Resul
 
     for unit in units {
         let dest = multiuser_wants.join(unit);
-        if rootfs_dfd.try_exists(&dest)? {
+        if rootfs_dfd
+            .symlink_metadata_optional(&dest)
+            .with_context(|| format!("Querying {unit}"))?
+            .is_some()
+        {
             continue;
         }
 
         println!("Adding {} to multi-user.target.wants", unit);
         let target = format!("/usr/lib/systemd/system/{unit}");
-        cap_primitives::fs::symlink_contents(target, &rootfs_dfd.as_filelike_view(), dest)?;
+        cap_primitives::fs::symlink_contents(target, &rootfs_dfd.as_filelike_view(), dest)
+            .with_context(|| format!("Linking {unit}"))?;
```

Comment:
Rather than `symlink_metadata_optional` + `symlink_contents`, can't we use `ensure_symlink` instead? It already knows to no-op if the destination exists.

Reply:
`ensure_symlink` is doing things related to directories as targets; also it requires that the existing path be a symlink but in theory I think the contents of a `wants.d` directory could be a regular file (though this would be discouraged).

But I agree we could probably factor out a helper here.  OTOH I did test this patch.